### PR TITLE
Fixing regression caused by WTForms

### PR DESCRIPTION
**Pull request by jonnyshaw89**
### What is the context of this PR?
Re-implementing feature in https://github.com/ONSdigital/eq-survey-runner/pull/897 that was regressed in https://github.com/ONSdigital/eq-survey-runner/pull/885

### How to review 
Re-test https://github.com/ONSdigital/eq-survey-runner/issues/883

### CODE REVIEW

File: app/forms/household_relationship_form.py [hunk 1]
```diff
@@ -17,7 +18,7 @@ def build_relationship_choices(answer_store, group_instance):
     :param group_instance: The instance of the group being iterated over
     :return:
     """
-    household_answers = answer_store.filter(block_id='household-composition')
+    household_answers = answer_store.filter(block_id='household-composition', limit=settings.EQ_MAX_NUM_REPEATS)
```
Comment (benjefferies) [inline app/forms/household_relationship_form.py]: Maybe a test to prevent regressions?

**Review by benjefferies — CHANGES_REQUESTED** [review]

File: app/forms/household_relationship_form.py [hunk 2]
```diff
@@ -17,7 +18,7 @@ def build_relationship_choices(answer_store, group_instance):
     :param group_instance: The instance of the group being iterated over
     :return:
     """
-    household_answers = answer_store.filter(block_id='household-composition')
+    household_answers = answer_store.filter(block_id='household-composition', limit=settings.EQ_MAX_NUM_REPEATS * 2)
```
Comment (iwootten) [inline app/forms/household_relationship_form.py]: Why x 2?

**Review by iwootten — CHANGES_REQUESTED** [review]

File: app/forms/household_relationship_form.py (hunk 2 above)
Comment (jonnyshaw89) [reply app/forms/household_relationship_form.py]: because it should have an answer for first-name and an answer for last-name

Comment (iwootten) [reply app/forms/household_relationship_form.py]: Won't it also return middle names? Maybe it would be better to query each answer_id individually?

Comment (collisdigital) [reply app/forms/household_relationship_form.py]: Last name isn't mandatory by the way

File: app/forms/household_relationship_form.py [hunk 3]
```diff
@@ -17,16 +18,17 @@ def build_relationship_choices(answer_store, group_instance):
     :param group_instance: The instance of the group being iterated over
     :return:
     """
-    household_answers = answer_store.filter(block_id='household-composition')
+    household_first_name_answers = answer_store.filter(block_id='household-composition', answer_id='first-name', limit=settings.EQ_MAX_NUM_REPEATS)
```
Comment (iwootten) [inline app/forms/household_relationship_form.py:21]: I guess the problem with this approach is that we can't guarantee that the last_names each match with the first_name as last_name is not mandatory.

**Review by iwootten — CHANGES_REQUESTED** [review]

File: app/forms/household_relationship_form.py (hunk 3 above)
Comment (jonnyshaw89) [reply app/forms/household_relationship_form.py:21]: That problem comes with the zip? would the answer-instance maintain the gaps?

Comment (iwootten) [reply app/forms/household_relationship_form.py:21]: Ah yes, but I think if I remember correctly we are storing blank answers for this particular form where there's nothing entered (the tests all seem to assume this), so probably not a problem.

Comment (collisdigital) [reply app/forms/household_relationship_form.py:21]: Assuming that change thats in another PR doesn't stop it storing blank entries, the trim thing?

Comment (iwootten) [reply app/forms/household_relationship_form.py:21]: Well, that will just strip it down to the empty string '' and is only for mandatory answers...so won't affect middle or last names. Middle and last names are stripped by the optional validator, but don't have to be populated and store blank entries. You can see some existing tests which haven't been altered in my branch for such behaviour here: https://github.com/ONSdigital/eq-survey-runner/blob/master/tests/app/forms/test_household_composition.py#L62 

I should note this behaviour is as it was prior to the WTForms merge.

**Review by iwootten — APPROVED** [review]

**Review by benjefferies — APPROVED** [review]

**Pull request merged**